### DISCUSSION TITLE
tls: server selected unsupported protocol version

### DISCUSSION THREAD

**Discussion by c-sh0** [projectdiscovery/httpx#633] [Q&A] [answered]
Ran into this issue and thought I'd share since I hadn't seen this mentioned anywhere. This is really a go issue and not a problem with httpx (per se?) httpx will not report this failure when connecting to a service with deprecated tls versions. As a result you may miss hosts running http services. You will only see this error when running httpx with the `-debug` or `-verbose` flag. 

`[DBG] Failed ...... tls: server selected unsupported protocol version 301` 

It's an easy workaround but, is set to be deprecated in Go 1.19. Set the GODEBUG environment variable when running httpx
`GODEBUG=tls10default=1`

Reference: 
https://tip.golang.org/doc/go1.18#tls10

Bug? Not a Bug?

**Comment by ehsandeep** [comment, marked as answer]
@c-sh0, thanks for opening this discussion; this is something we are already tracking internally https://github.com/projectdiscovery/team-backlogs/issues/156 since the release of **Go 1.18**, and something we will be working on as this impacts multiple projects we have published.

**Comment by ehsandeep** [comment]
@c-sh0 https://github.com/projectdiscovery/httpx/pull/645
